onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -191,15 +191,29 @@ bool ReshapeFusion::Match_One_Element_Output_Subgraph_2(Graph& graph, const Node
 
     // Check if Slice op slices 1d array (result of shape) to one element.
     std::vector<int64_t> slice_inputs;
-    if (optimizer_utils::AppendTensorFromInitializer(graph, *(slice.InputDefs()[1]), slice_inputs, true) &&
-        optimizer_utils::AppendTensorFromInitializer(graph, *(slice.InputDefs()[2]), slice_inputs, true)) {
-      const int64_t slice_start = slice_inputs[0];
-      const int64_t slice_end = slice_inputs[1];
-      if (!(slice_end >= INT_MAX && slice_start == -1) && abs(slice_end - slice_start) != 1) {
-        return false;
+    int64_t slice_start(0);
+    int64_t slice_end(0);
+    if (slice.GetInputEdgesCount() >= 3) {
+      if (optimizer_utils::AppendTensorFromInitializer(graph, *(slice.InputDefs()[1]), slice_inputs, true) &&
```

Comment:
What happens here if starts=[0, 0] (two elements instead of one)? Maybe you can fuse some of the logic here with the below and have a std::vector for starts/ends that each InputDef goes into and then the size() check from below would catch this? It would also shrink the code because there would be one less std::vector to construct/destruct.

Reply:
BTW, I saw that MatchGemmSubgraph in attention_fusion_helper.h also takes Slice-1 but CheckSliceParameters doesn't deal with this. Should that path just ignore Slice-1 instead?

Reply:
There should be a test added for this too.

Reply:
I think that the current implementation ignores the match of old version slice. We could support slice-1 for all matching,  or just ignore it but ensure that there shall be no core dump. What do you think?